_posts/2019/2019-09-20-linux-party-experience.md

```diff
@@ -0,0 +1,158 @@
+
+[This](https://www.youtube.com/watch?v=HbgzrKJvDRw) is a great video explaining the Linux filesystem in detail and don't worry if a lot of things go over your head, for starters, we are only concerned with /, /home and swap
+
+/home is where a user's files are stored(similar to the C:\Users in windows) but you would have to allocate more space than what you might be using in C:\Users because ALL of the user files will be in /home and there would be no other D: or E: drive to store these files
+
+swap is a 'virtual RAM' that is utilised to extend RAM beyond its physical capacity, with swap memory an 8 GB RAM can 'store' much more data than 8 GB though it would be much slower.
+
+Now for all this blabbering from our experience a general space ratio of 40:60 between / and /home and having swap as double the size of your RAM is the ideal composition. Also, root requires a minimum of 20 GB for smooth operation
+
+## The actual partitioning
+
+Now we have three cases
+
+* Only SSD or only HDD
+* SSD + HDD
+* Optane + HDD
+
+Now for only HDD laptops, the work is pretty straightforward that you have to shrink the required memory which you have calculated previously. 
+
+For only SSD the process is the same as only HDD laptops except that you have to shift your SSD from RAID to AHCI using [this](https://support.thinkcritical.com/kb/articles/switch-windows-10-from-raid-ide-to-ahci). Step 5 of the previous link will vary with manufacturer, so explore your BIOS a bit (don't go on changing any random setting!) and you'll be bound to hit it or you can google for "____ RAID to AHCI".
+
+For SSD + HDD laptops you would want to keep minimum stuff on SSD as it is generally small. So what should be done in this case is to allocate space for / on the SSD(minimal boot time) and /home on the HDD.
+
+For shrinking your Windows drives follow [this guide](https://www.dummies.com/computers/pcs/how-to-shrink-a-hard-drive-volume-in-windows/) for both SSD and HDD
+
+Optane is rare so first check [this](https://www.intel.in/content/www/in/en/support/articles/000023990/memory-and-storage/intel-optane-memory.html) to determine if you have Optane in use in case you are unsure. Installing Ubuntu on Optane is a bit complex so we won't be going there, use [this](https://support.cyberpowerpc.com/hc/en-us/articles/360014775073-How-do-I-disable-Intel-Optane-)(also Disable Intel Optane memory with Intel RST is enough, no need disable from BIOS!), install ubuntu but remember to keep EXTRA 10 MB unallocated space at the END of your HDD which is required by Optane, other than the memory used by Ubuntu. Then use the previous guide to reenable Optane.
```

Comment:
```suggestion
Optane is rare so first check [this](https://www.intel.in/content/www/in/en/support/articles/000023990/memory-and-storage/intel-optane-memory.html) to determine if you have Optane in use in case you are unsure. Installing Ubuntu on Optane is a bit complex so we won't be going there; use [this](https://support.cyberpowerpc.com/hc/en-us/articles/360014775073-How-do-I-disable-Intel-Optane-)(disabling Intel Optane memory with Intel RST is enough, no need disable from BIOS!) and install Ubuntu, but remember to keep an EXTRA 10 MB unallocated space at the END of your HDD which is required by Optane, other than the memory used by Ubuntu. Then use the previous guide to re-enable Optane.
```